xray/segment.go

```diff
@@ -638,3 +651,38 @@ func (seg *Segment) addError(err error) {
 	seg.GetCause().WorkingDirectory, _ = os.Getwd()
 	seg.GetCause().Exceptions = append(seg.GetCause().Exceptions, seg.ParentSegment.GetConfiguration().ExceptionFormattingStrategy.ExceptionFromError(err))
 }
+
+// TraceHeaderID return trace id for http header
+func (seg *Segment) TraceHeaderID(traceHeader *header.Header) string {
+	seg.Lock()
+	defer seg.Unlock()
+
+	var respHeader bytes.Buffer
+	respHeader.WriteString("Root=")
+	respHeader.WriteString(seg.TraceID)
+
```

Comment:
Same comment for `TraceHeaderId` and `HTTPCapture` methods. Any specific reason for bringing this here? We can also add these methods inside `handler.go` file. Also, not sure if we want to expose this APIs, this can be better served as internal helper methods I think.

Reply:
The reason for being in this file is because they are methods of the `Segment` structure, so I thought it would make sense to be in `segment.go`.

I'm changing this and I'm going to create helper functions in `handler.go`.